autometrics/src/prometheus_exporter.rs

```diff
@@ -2,47 +2,89 @@ use crate::HISTOGRAM_BUCKETS;
 #[cfg(feature = "metrics")]
 use metrics_exporter_prometheus::{PrometheusBuilder, PrometheusHandle};
 use once_cell::sync::Lazy;
+#[cfg(feature = "opentelemetry")]
 use opentelemetry_prometheus::{exporter, PrometheusExporter};
+#[cfg(feature = "opentelemetry")]
 use opentelemetry_sdk::export::metrics::aggregation;
+#[cfg(feature = "opentelemetry")]
 use opentelemetry_sdk::metrics::{controllers, processors, selectors};
-use prometheus::{default_registry, Error, TextEncoder};
+#[cfg(any(feature = "opentelemetry", feature = "prometheus"))]
+use prometheus::TextEncoder;
+use thiserror::Error;
 
-static GLOBAL_EXPORTER: Lazy<GlobalPrometheus> = Lazy::new(|| initialize_metrics_exporter());
+#[derive(Debug, Error)]
+pub enum EncodingError {
+    #[cfg(any(feature = "prometheus", feature = "opentelemetry"))]
+    #[error(transparent)]
+    Prometheus(#[from] prometheus::Error),
+    #[cfg(feature = "prometheus-client")]
+    #[error(transparent)]
+    Format(#[from] std::fmt::Error),
+}
+
+static GLOBAL_EXPORTER: Lazy<GlobalPrometheus> = Lazy::new(|| GlobalPrometheus {
+    #[cfg(feature = "metrics")]
+    metrics_exporter: PrometheusBuilder::new()
+        .set_buckets(&HISTOGRAM_BUCKETS)
+        .expect("Failed to set histogram buckets")
+        .install_recorder()
+        .expect("Failed to install recorder"),
```

Comment:
is it really appropriate for a library to panic? it'd be better if we just returned `Result`s so end user programs don't end up panic-ing